packages/cli/src/ui/components/InputPrompt.tsx

```diff
@@ -285,6 +289,35 @@ export const InputPrompt: React.FC<InputPromptProps> = ({
         return;
       }
 
+      // Check if this is the special backslash+return sequence
+      const isPrevBackslash =
```

Comment:
can we handle this up a level rather than input prompt? this seems simpler to handle as part of processing keyboard events that directly in InputPrompt.  that could avoid the currently logic with prevKeyRef ideally surfacing this just as shift+enter as far as inputPrompt is concerned.

Reply:
Do you think doing this in the keypress hook would be better?

Reply:
Ive moved it to keypress hook.